modules/combined/examples/periodic_strain/global_strain_pfm.i

```diff
@@ -0,0 +1,424 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 50
+  ny = 50
+  xmin = -0.5
+  xmax = 0.5
+  ymin = -0.5
+  ymax = 0.5
+[]
+
+[MeshModifiers]
+  [./cnode]
+    type = AddExtraNodeset
+    coord = '0.0 0.0'
+    new_boundary = 100
+  [../]
+[]
+
+[Variables]
+  [./u_x]
+  [../]
+  [./u_y]
+  [../]
+  [./global_strain]
+    order = THIRD
+    family = SCALAR
+  [../]
+  [./c]
+    [./InitialCondition]
+      type = FunctionIC
+      function = 'sin(2*x*pi)*sin(2*y*pi)*0.05+0.6'
+    [../]
+  [../]
+  [./w]
+  [../]
+[]
+
+[AuxVariables]
+  [./dispg_x]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./dispg_y]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./local_energy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./disp_x]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./disp_y]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./s00]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./s01]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./s10]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./s11]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./e00]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./e01]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./e10]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./e11]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[AuxKernels]
+  [./ug_x]
+    type = GlobalDisplacementAux
+    variable = dispg_x
+  [../]
+  [./ug_y]
+    type = GlobalDisplacementAux
+    variable = dispg_y
+    component = 1
+  [../]
+  [./u_x]
+    type = TotalDisplacementAux
+    variable = disp_x
+    displacement_variables = 'u_x dispg_x'
```

Comment:
I think I'd prefer if there was a single AuxKernel to get the total displacement without having the extra dispg_x/y variables. TotalDisplacementAux really is just a shorthand for a ParseAux that adds two variables.

Reply:
Okay, I will combine both operations in single AuxKernel.